### PR TITLE
Ft(update-profile): Update user profile

### PR DESCRIPTION
**Pull request by abraham-kwizera**
## What does this PR do?

> Preview Current User Data and Update profile 

## Description of Task to be completed?

- [x] Added Skeleton to the Form 
- [x]  Add _Success/Unsuccessful Attempt_ messages 
- [x] fetch and update user data integration
- [x] Created a **view profile** page for displaying user's info before edit
- [x] Created an **update profile** page for implementing updates if the user wishes, which appears when clicked on the update profile button
- [x] Implemented FETCH and PATCH for manipulation of data relating to current user state managed by redux
- [x] Updated a build script to ensure that all dependencies in package.json are passed during installation

## How should this be manually tested?

1. clone git repo, if not yet done 
2. `cd Avengers-Phantom-FE` 
3. `npm i` or `npm install` to install dependencies 
4. `npm run dev` to start the server 
5. On a loaded **Homepage**, click on **Sign in** button in the menu bar
6. Enter valid credentials given by the System administrator 
7. After a valid sign-in attempt, you are redirected  to the main wall of the dashboard, click on the **Profile** tab in. ready to manipulate your data  :rocket:

**Sample Image description:** 
_1/3: Preview current user data:_
Screen Shot 2022-06-24 at 00 54 13

_2/3: Form data manipulation:_ 
Screen Shot 2022-06-24 at 00 55 44

_3/3: Data after updates:_
Screen Shot 2022-06-24 at 00 56 04

END

### CODE REVIEW

**Review by Byadiso — APPROVED** [review]

**Comment by Anguandia** [comment]
Hi @abraham-kwizera, good work! please take note of the following:
- Squash your commits to 1 when possible
- It would be good to add some screenshots to the description
- Liaise with @Gedeonir to get the issue of the failing build resolved

File: src/Components/Homepage/MapSection.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import Card from "@mui/material/Card";
 import Button from "@mui/material/Button";
 import Menu from "@mui/material/Menu";
 import { NavigationFormValidator } from "./NavigationFormValidator.jsx";
-import { MapContainer, TileLayer, useMap, Marker, Popup } from "react-leaflet";
+// import { MapContainer, TileLayer, useMap, Marker, Popup } from "react-leaflet";
```
Comment (Anguandia) [inline src/Components/Homepage/MapSection.jsx]: Are you sure these are not needed? please consult with @Jeanndo before commenting the lines out

File: src/Components/Homepage/MapSection.jsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import Card from "@mui/material/Card";
...
 
 // console.log(TeslaData);
```
Comment (Anguandia) [inline src/Components/Homepage/MapSection.jsx]: Please delete such commented-out lines if sure they are not needed

File: src/Components/signin/Signin.jsx [hunk 3]
```diff
@@ -87,7 +87,7 @@ export const Signin = () => {
           console.log(data);
```
Comment (Anguandia) [inline src/Components/signin/Signin.jsx]: remove all unnecessary console.log statements

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work @abraham-kwizera , 
- please address the few change requests suggested,
- improve the PR description, especially on how to test after starting the server; which pages to navigate to, what actions to perform, etc
- a couple of screenshots would go a long way in enriching the PR description
- then we'll be good to go

**Review by Anguandia — APPROVED** [review]

**Pull request merged**